src/responses.js

```diff
@@ -0,0 +1,179 @@
+    if (openApi3 && hasFileRef(body)){
+        return getRefFromInclude(body)
+    }
+    if (!header.value.match(/application\/.*json/)) {
+        return body
+    } 
+    try {
+        return JSON.parse(body);
+    } catch (e) { }
+}
+
+const getResponseSchema = (response, options) => {
+    const { preferReference, openApi3 } = options
+    const componentsPath = openApi3 ? '#/components/schemas' : '#/definitions/'
+    if (preferReference) { // MSON then schema
+        const inputSchema = searchDataStructure(response.content, openApi3); // Attributes in response
+        if (inputSchema) {
+            return inputSchema
+        } else if (response.reference) {
+            return {
+                '$ref': componentsPath + escapeJSONPointer(response.reference.id + 'Model')
+            };
+        } else if (response.schema) {
+            return parseResponseSchema(response.schema, openApi3)
+        }
+    } else { // schema then MSON
+        if (response.schema) {
+            return parseResponseSchema(response.schema, openApi3)
+        }
+       
+        const inputSchema = searchDataStructure(response.content, openApi3); // Attributes in response
+        if (inputSchema) return inputSchema;
+        if (response.reference) {
+            return {
+                '$ref': componentsPath + escapeJSONPointer(response.reference.id + 'Model')
+            };
+        }
+    }
+}
+
+const setResponseSchema = (responses, response, schema, openApi3) => {
+    if (!openApi3){
+        responses[response.name].schema = schema
+        return responses
+    }
+    // In openAPI 3 the schema lives under the content type
+    const contentTypeHeader = response.headers.find((h) => h.name.toLowerCase() === 'content-type')
+    if (!contentTypeHeader.value) {
+        return responses
+    }
+    if (!responses[response.name].content[contentTypeHeader.value]){
+        responses[response.name].content[contentTypeHeader.value] = {}
+    }
+    
+    if (!responses[response.name].content[contentTypeHeader.value].schema){
+        responses[response.name].content[contentTypeHeader.value].schema = schema
+        return responses
+    }
+
+    // If a schema already exists, we need to use oneOf for additional unique schemas.
+    let existingSchema = { ...responses[response.name].content[contentTypeHeader.value].schema }
+    // It is possible that the given schema is a duplicate. If that's the case, we don't add it.
+    if (isEqual(existingSchema, schema)){
+        return responses
+    }
+
+    if (existingSchema.oneOf){
+        if (!existingSchema.oneOf.find((s) => isEqual(s, schema))){
+            responses[response.name].content[contentTypeHeader.value].schema.oneOf.push(schema)
+        }
+    } else {
+        responses[response.name].content[contentTypeHeader.value].schema = { oneOf: [existingSchema, schema] }
+    }
+    return responses
+}
+
+const setResponseExample = (responses, response, header, body, openApi3) => {
```

Comment:
Instead of explicitly passing in `openapi3`, let's use the `options` variable or `context` to derive the state